src/node/entities.h

```diff
@@ -60,50 +60,68 @@ namespace ccf
 
   struct Tables
   {
-    // Governance tables
+    // Service tables
+
+    // Member identities
     static constexpr auto MEMBERS = "public:ccf.gov.members";
-    static constexpr auto MEMBER_ACKS = "public:ccf.gov.member_acks";
-    static constexpr auto MEMBER_CERT_DERS = "public:ccf.gov.member_cert_ders";
-    static constexpr auto MEMBER_DIGESTS = "public:ccf.gov.member_digests";
+    static constexpr auto MEMBER_ACKS = "public:ccf.gov.members.acks";
+    static constexpr auto MEMBER_CERT_DERS =
+      "public:ccf.internal.members.certs_der";
+    static constexpr auto MEMBER_DIGESTS =
+      "public:ccf.internal.members.digests";
+
+    // User identities
     static constexpr auto USERS = "public:ccf.gov.users";
-    static constexpr auto USER_CERT_DERS = "public:ccf.gov.user_cert_ders";
-    static constexpr auto USER_DIGESTS = "public:ccf.gov.user_digests";
+    static constexpr auto USER_CERT_DERS =
+      "public:ccf.internal.users.certs_der";
+    static constexpr auto USER_DIGESTS = "public:internal.gov.users.digests";
     static constexpr auto SERVICE_PRINCIPALS =
```

Comment:
Quite a general thought, but raising it here:

We want to allow apps to define their own proposal actions, on their own tables. This is actually an instance of a table which should really be in the app, not the framework. I think the JS application tables are the same - they shouldn't be in every app. And if they're not in every app, I don't think they should be `ccf.` prefixed - the point is to reserve that for names which the _service_ needs, and apps should _not_ be allowed to modify any `ccf.` tables. So if we're trying to get to final naming, maybe we should remove `ccf.` from these? And then I'm not sure whether it makes sense to keep `gov.` - its probably still a helpful distinction, but is it a guideline or a requirement for app proposed tables?

Reply:
For this specific table (`SERVICE_PRINCIPALS`), my understanding is that it will be removed once we move to JS governance and dynamic tables are supported (what you described above). Renaming it to `public:ccf.app.service_principals` makes sense though.

Are you also referring to the 3 JS tables (scripts, modules and endpoints)? If yes, it seems that they are very much hardcoded in the member frontend to be able to bootstrap a JS app. I've also prefixed them with `public:ccf.app.`.

Reply:
They're in `member_frontend` currently, but those proposals don't need to be there anymore than `SERVICE_PRINCIPALS` (well maybe `modules` is reused for governance modules, but the gist remains). The proposals could be defined in the governance script, or even in `js_generic.cpp`. Eventually, hopefully, ideally.

Reply:
Good points. I wonder whether they should still be prefixed (either `public:ccf.app` or `public:app.meta`) so that they don't clash with app-defined tables though? I believe we don't want app endpoints to read/write in these tables?

Reply:
I'd lean towards giving them a `gov.` prefix. So proposals can write to (and are the _only_ thing that can write to) `gov.` and `ccf.gov.`. Then apps can't write to these tables because they're "governance", rather than because they're CCF.

Reply:
Using `gov.` prefix now.